### PR TITLE
Fix 1353: tab navigation via Options menu

### LINKED ISSUE #1353: Cannot change tabs on Windows
**Describe the bug**
I cannot change tabs, Lazygit only crashes. This is on Windows 10, Lazygit 0.28.1. Git-bash and WSL. Installed from Scoop.

**To Reproduce**
Change tabs via the `x` menu.

**Expected behavior**
It should change tabs.

**Screenshots**
example

**Desktop (please complete the following information):**
 - OS: Windows
 - Git bash and WSL2
 - Lazygit Version 0.28.1

**Additional context**
Error message:
```
panic: runtime error: integer divide by zero

goroutine 1 [running]:
github.com/jesseduffield/lazygit/pkg/utils.ModuloWithWrap(...)
        /home/runner/work/lazygit/lazygit/pkg/utils/utils.go:273
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).handleNextTab(0xc0001561c0, 0xc00011d380, 0xc0005fe010)
        /home/runner/work/lazygit/lazygit/pkg/gui/view_helpers.go:368 +0x16a
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).handleCreateOptionsMenu.func1(0xc0001561c0, 0x0)
        /home/runner/work/lazygit/lazygit/pkg/gui/options_menu_panel.go:69 +0x79
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).onMenuPress(0xc0001561c0, 0x0, 0x0)
        /home/runner/work/lazygit/lazygit/pkg/gui/menu_panel.go:101 +0x92
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).wrappedHandler.func1(0xc00011d380, 0xc000003e00, 0x88bf00, 0xc0003d58e0)
        /home/runner/work/lazygit/lazygit/pkg/gui/confirmation_panel.go:314 +0x2d
github.com/jesseduffield/gocui.(*Gui).execKeybinding(0xc00011d380, 0xc000003e00, 0xc00035be40, 0x10100000000, 0x0, 0x2)
        /home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:1167 +0x6d
github.com/jesseduffield/gocui.(*Gui).execKeybindings(0xc00011d380, 0xc000003e00, 0xc0003d5900, 0xc00004c000, 0xc0003d58f0, 0xc000066780)
        /home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:1135 +0x1d1
github.com/jesseduffield/gocui.(*Gui).onKey(0xc00011d380, 0xc0003d5900, 0xc00005ac10, 0xc000116200)
        /home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:1060 +0x2ce
github.com/jesseduffield/gocui.(*Gui).handleEvent(0xc00011d380, 0xc0003d5900, 0x0, 0x0)
        /home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:651 +0x4d
github.com/jesseduffield/gocui.(*Gui).MainLoop(0xc00011d380, 0xc000000004, 0xc0003f1a08)
        /home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:611 +0x1d8
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).Run(0xc0001561c0, 0x0, 0x0)
        /home/runner/work/lazygit/lazygit/pkg/gui/gui.go:540 +0x51b
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError.func1(0xc0002f8001, 0xc0002d5aa0)
        /home/runner/work/lazygit/lazygit/pkg/gui/gui.go:548 +0x48
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0xc0000a3b40, 0x0, 0x0)
        /home/runner/work/lazygit/lazygit/pkg/utils/utils.go:380 +0x67
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError(0xc0001561c0, 0x0, 0xfcd9d0)
        /home/runner/work/lazygit/lazygit/pkg/gui/gui.go:547 +0x88
github.com/jesseduffield/lazygit/pkg/app.(*App).Run(0xc0002d79d0, 0xc0002be5a0, 0x0)
        /home/runner/work/lazygit/lazygit/pkg/app/app.go:242 +0x73
main.main()
        /home/runner/work/lazygit/lazygit/main.go:127 +0xb9b
```

### PR DESCRIPTION
**Pull request by Krismix1**
Fix #1353 when attempting to change tabs via the Option menu. The description of `ParentView` remotely matches the behaviour needed, but perhaps it was initially intended to be used differently.

Open for feedback.

### CODE REVIEW

File: pkg/gui/view_helpers.go [hunk 1]
```diff
@@ -392,3 +392,17 @@ func (gui *Gui) pageDelta(view *gocui.View) int {
 
 	return delta
 }
+
+func getTabbedView(gui *Gui) *gocui.View {
```
Comment (jesseduffield) [inline pkg/gui/view_helpers.go:396]: I added the ParentView field to my gocui fork a while back, before I introduced the concept of Contexts. I haven't officially deprecated it but it is in essence deprecated in favour of Contexts. We have a Context stack that typically starts at the bottom with the current side view and in this case ends at the top with our menu view. The `gui.currentStaticContext()` function starts at the top of the stack and goes down until it finds a static (i.e. non-popup) context.

I reckon we should do the following:
```go
func getTabbedView(gui *Gui) *gocui.View {
	// assumes only static contexts have tabs which is an assumption I consider safe
	context := gui.currentStaticContext()
	view, _ := gui.g.View(context.GetViewName())
	return view
}
```

**Review by jesseduffield — CHANGES_REQUESTED** [review]
Looking good! Also great work on the integration test. I've taken the liberty of pushing a slight tweak to it so that it's less brittle: we're now searching for the item by name in the menu rather than moving our cursor all the way to it, so that if we add more items to the menu down the line, the test won't break.

**Comment by jesseduffield** [comment]
Looks like you'll also need to rebase onto the master branch to fix that linting issue

File: pkg/gui/view_helpers.go (hunk 1 above)
Comment (Krismix1) [reply pkg/gui/view_helpers.go:396]: Updated to static contexts, and removed the `ParentView` assignment to not add code using a deprecated field.

File: pkg/gui/menu_panel.go [hunk 2]
```diff
@@ -74,11 +74,13 @@ func (gui *Gui) createMenu(title string, items []*menuItem, createMenuOptions cr
 
 	list := utils.RenderDisplayStrings(stringArrays)
 
+	parentView := gui.g.CurrentView()
 	x0, y0, x1, y1 := gui.getConfirmationPanelDimensions(false, list)
 	menuView, _ := gui.g.SetView("menu", x0, y0, x1, y1, 0)
 	menuView.Title = title
 	menuView.FgColor = theme.GocuiDefaultTextColor
 	menuView.ContainsList = true
+	menuView.ParentView = parentView
```
[the remark this answers is not in the document]
Comment (Krismix1) [reply pkg/gui/menu_panel.go]: Updated to static contexts, and removed the `ParentView` assignment to not add code using a deprecated field.

**Comment by Krismix1** [comment]
I have updated the changes to use the static contexts. Good idea with using the search instead of hardcoded navigation.
I have rebased the branch, let's this pipeline be green.

**Comment by jesseduffield** [comment]
nice one :)

**Pull request merged**